### DISCUSSION TITLE
getting [ERROR] KeyError: 'http'

### DISCUSSION THREAD

**Discussion by lalitbyteiq** [aws-powertools/powertools-lambda-python#5163] [Q&A]
im getting bellow exception with rest api 
![image](https://github.com/user-attachments/assets/fffed5d0-c5f5-4bd7-bb90-89866e0b695e)

but whilke using http api in api gateway its working 

sample code :

from aws_lambda_powertools import Logger, Tracer
from aws_lambda_powertools.event_handler import APIGatewayHttpResolver
from aws_lambda_powertools.logging import correlation_paths
from aws_lambda_powertools.utilities.typing import LambdaContext
from pydantic import BaseModel, Field

tracer = Tracer()
logger = Logger()
app = APIGatewayHttpResolver()

class InputRequest(BaseModel):  
    userId: int
    title: str
    completed: bool = None

@app.get("/ping")
@tracer.capture_method
def get_ping():

    return {"todos": "live", "status": "200"}
    
@app.get("/todo/{task}")
@tracer.capture_method
def get_todos(task: str):
    return {"todos": f"todo has been included {task}", "status": "200"}
    
    
@app.post("/post-test/")
@tracer.capture_method
def get_todos(input: InputRequest):
    return input.model_dump()
    
    

# You can continue to use other utilities just as before
@logger.inject_lambda_context(correlation_id_path=correlation_paths.API_GATEWAY_HTTP)
@tracer.capture_lambda_handler
def lambda_handler(event: dict, context: LambdaContext) -> dict:
    print(event)
    return app.resolve(event, context)

**Comment by leandrodamascena** [comment]
Hi @lalitbyteiq! Thanks for opening this discussion. I'm quite confused about the scenario here and please help me understand it better.

If I understand correctly, when you are using the above code, things are working as expected with Amazon API Gateway HTTP, right? But you mentioned that when using RestAPI (in Amazon API Gateway?) it crashes. Is that correct? If yes, it's because AWS send [different payloads](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html) to Lambda when integrating Lambda with Rest/HTTP API. If you want to integrate RestAPI with Lambda and use Powertools, please use [APIGatewayRestResolver](https://docs.powertools.aws.dev/lambda/python/latest/core/event_handler/api_gateway/#api-gateway-rest-api) resolver.

Please let me know if you need additional help.

**Reply by lalitbyteiq** [reply]
Hay sorry my bad i have used http resolver instead of rest.

thanks

**Reply by leandrodamascena** [reply]
Yeah, you need to use the Rest resolver. Please let me know if this is working now.
